src/glib/base/hash.h

```diff
@@ -1088,21 +1086,26 @@ void TCache<TKey, TDat, THashFunc>::Purge(const int64& MemToPurge){
 
 template <class TKey, class TDat, class THashFunc>
 int64 TCache<TKey, TDat, THashFunc>::GetMemUsed() const {
-	int64 MemUsed = 2 * sizeof(int64);
-	
-    MemUsed += KeyDatH.GetMemUsed(false);
-    MemUsed += TimeKeyL.GetMemUsed();
-	int cnt = 0;
-    int KeyId = KeyDatH.FFirstKeyId();
-    while (KeyDatH.FNextKeyId(KeyId)) {
-		const TKeyLNDatPr& KeyLNDatPr = KeyDatH[KeyId];
-		TDat Dat = KeyLNDatPr.Val2;
-		MemUsed += int64(Dat->GetMemUsed());
-		cnt++;
-	}   
-	EAssert(cnt == KeyDatH.Len());
-
-	return MemUsed;
+    return sizeof(TCache) +
+           TMemUtils::GetExtraMemberSize(MxMemUsed) +
+           TMemUtils::GetExtraMemberSize(CurMemUsed) +
+           TMemUtils::GetExtraContainerSizeShallow(KeyDatH) +
```

Comment:
while loop in the old version was also doing GetMemUsed on members of KeyDatH, we should use `GetExtraContainerSize(KeyDatH)` here instead of shallow version.